docs/kubernetes/user-guides/snapshots.md

```diff
@@ -19,33 +21,102 @@ $ ./deploy/kubernetes/deploy-driver.sh
 
 ### Snapshot Example
 
-1. Create example Default Snapshot Class
+**Create Storage Class:**
+
+If you haven't created a `StorageClass` yet, create one first:
+
+```console
+kubectl apply -f ./examples/kubernetes/snapshot/storageclass.yaml
 ```
-$ kubectl create -f ./examples/kubernetes/demo-defaultsnapshotclass.yaml
+
+For more advance `StorageClass` configuration, please see [Kubernetes Basic User Guide](/docs/kubernetes/user-guides/basic.md).
+
+**Create  Default VolumeSnapshotClass:**
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/default_volumesnapshotclass.yaml
 ```
-2. Create a snapshot of the PVC created in above example
+
+**Create source PVC:**
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/source_pvc.yaml
+```
+
+**Generate sample data:**
+
+Create a sample pod with the source PVC. The source PVC is mounted into `/demo/data` directory of this pod.
+
+```console
+kubectl create -f ./examples/kubernetes/snapshot/source_pod.yaml
+```
+
+Now, let's create a file inside `/demo/data` directory:
+
+```console
+kubectl exec source-pod -- touch /demo/data/sample-file.txt
```

Comment:
I think maybe the `source-pod` busybox can create this file without having an extra exec step here

Reply:
Removed this step. Now `source-pod` generate sample file automatically.